drivers/entropy/entropy_stm32.c

```diff
@@ -152,6 +158,30 @@ static int entropy_stm32_rng_init(struct device *dev)
 	__ASSERT_NO_MSG(dev_data != NULL);
 	__ASSERT_NO_MSG(dev_cfg != NULL);
 
+#if CONFIG_SOC_SERIES_STM32L4X
+	/* Configure PLLSA11 to enable 48M domain */
+	LL_RCC_PLLSAI1_ConfigDomain_48M(LL_RCC_PLLSOURCE_MSI,
+					LL_RCC_PLLM_DIV_1,
+					24, LL_RCC_PLLSAI1Q_DIV_2);
+
+	/* Enable PLLSA1 */
+	LL_RCC_PLLSAI1_Enable();
+
+	/*  Enable PLLSAI1 output mapped on 48MHz domain clock */
+	LL_RCC_PLLSAI1_EnableDomain_48M();
+
+	/* Wait for PLLSA1 ready flag */
+	while(LL_RCC_PLLSAI1_IsReady() != 1)
```

Comment:
I can't comment on the STM32 specifics, but there are stylistic issues with this `while` statement here: there must be a space between the keyword and the opening parenthesis; the opening brace should be in the same line as the `while`, and there shouldn't be a semicolon after the closing bracket.

What's the usual time it takes for the PLLSA1 flag to be set? This could impact early boot time since device initialization happens serially.

Reply:
@lpereira: Fixing styling issue. Thanks!
About PLLSAI flag, PLLSAI Lock time is max 40µs, typical 15µs. Is that blocking?
If we want RNG, this anyway required. Initialisation level is PRE_KERNEL_2, not sure if it would make sense to change it.

Reply:
40 microseconds boot time increase in the worst case scenario is a good price to pay for a HWPRNG.